File: Sources/PhotoPullCore/ImportResult.swift
import Foundation

/// Итог операции импорта — используется для отображения сводки в UI и в тестах.
public struct ImportResult: Equatable, Sendable {
    public var succeeded: Int
    public var failed: Int

    public init(succeeded: Int = 0, failed: Int = 0) {
        self.succeeded = succeeded
        self.failed = failed
    }

    public var total: Int { succeeded + failed }

    public var isEmpty: Bool { total == 0 }

    /// Короткое человекочитаемое описание для UI.
    public var summary: String {
        if isEmpty {
            return "Нет файлов для импорта"
        }
        if failed == 0 {
            return "Импортировано файлов: \(succeeded)"
        }
        return "Импортировано: \(succeeded), с ошибкой: \(failed)"
    }
}

File: shared/src/commonMain/kotlin/com/async/shared/ImportResult.kt
package com.async.shared

/**
 * Итог операции импорта — используется для отображения сводки в UI и в тестах.
 */
data class ImportResult(
    val succeeded: Int = 0,
    val failed: Int = 0
) {
    val total: Int get() = succeeded + failed

    val isEmpty: Boolean get() = total == 0

    /** Короткое человекочитаемое описание для UI. */
    val summary: String
        get() = when {
            isEmpty -> "Нет файлов для импорта"
            failed == 0 -> "Импортировано файлов: $succeeded"
            else -> "Импортировано: $succeeded, с ошибкой: $failed"
        }
}
